### DISCUSSION TITLE
Why not async?

### DISCUSSION THREAD

**Discussion by asdzd029123asdssa231** [khonsulabs/gooey#95] [Ideas]
This is a small question, but why is this not lib async?

**Comment by ecton** [comment]
In short, I can't think of a reason for it to be, it would only add overhead, and I would have to pick a runtime or support flags for switching between them.

The Dynamic type is async-friendly. Async tasks can set values, read values, and even wait for the next value.

The current approach I'm using in a little client-server app I've started building is having a `fn tokio() -> &'static tokio::runtime::Handle` function that returns a handle to a running tokio runtime. This allows me to spawn tasks inside of my Gooey code, and I pass those async tasks the necessary Dynamic values.

If Rust ever supports function coloring where Gooey could support both async and non-async code paths, I would reconsider adding async at that point. Right now, I can't think of any reasons to add it, so I will finish my answer with a question: why should this library be async?
